### DISCUSSION TITLE
Directions

### DISCUSSION THREAD

**Discussion by darcymason** [darcymason/egsnrc#21] [Ideas]
Welcome everyone.

This discussion is to try to communicate some general directions for this project, and how to pair the technical aspects with what user groups need.

Discussion started in https://github.com/nrc-cnrc/EGSnrc/discussions/658 and moved here...

**Comment by darcymason** [comment]
I posted in the other discussion about starting a video chat ... for me the earliest date would be Wednesday (March 1), but Thursday 2nd or Friday 3rd would be better.

I've opened a [doodle poll](https://doodle.com/meeting/participate/id/ejYmNB4d) for possible times - I've suggested a few, if those don't work I can repeat with more options. 

The meeting is just to continue the discussion, get some ideas on what the scope of this project could/should be (e.g. CPU-capable vs GPU-only, priorities for ease-of-use vs. optimizing GPU, etc.).  I can also give some more details about what I've learned so far and what I think some of the technical challenges might be.

**Reply by darcymason** [reply]
Oh, one more thing: I put one-hour blocks in, but suspect that even 20 min could be a good start, perhaps 30 min with any technical issues in getting interested people on the chat.

**Reply by ftessier** [reply]
I've answered the poll. If you can manage meeting earlier in the day, it would help. We have regular EGSnrc future development meeting on Friday, so perhaps talking on Thursday would be best. Or on Friday morning 10:00 _in lieu_ of our meeting, if that's an option at your end.

**Reply by darcymason** [reply]
Mornings can work for me. I've started with afternoon because that could work for Simon, and I'd like to include him if he's interested.

**Reply by ftessier** [reply]
Yes, let's ensure we meet with Simon, ideally. So let's go with your Doodle times!

**Reply by SimonBiggs** [reply]
I filled out the doodle :)

**Reply by darcymason** [reply]
Based on the doodle poll I would suggest Thursday at 4 pm Eastern - @SimonBiggs , I'm assuming that's a little better for you as not so early.  But @ftessier, you had preferred earlier if possible, so we could go to 3 pm if needed.  Thoughts?

**Reply by SimonBiggs** [reply]
I'd prefer 4pm certainly (my 8 am) but can also do 3pm if need be (my 7am)

**Reply by ftessier** [reply]
16:00 is fine by me. Who sends the calendar or zoom invite? I will also ask @rtownson to join if he can, if that is ok with you.

**Reply by darcymason** [reply]
I thought I'd post the link here just before the time.  Do you have a preference for zoom/MS Teams/ Google Meet?  I'd probably use google meet if that works, but can do any of them.

And yes, @rtownson or anyone else interested is welcome.

**Reply by rtownson** [reply]
Sure, 4pm tomorrow works for me.

**Reply by SimonBiggs** [reply]
> I thought I'd post the link here just before the time.

Just wanting to check I have my timezones correct, we're meeting in 15 mins yes?

**Reply by darcymason** [reply]
@ftessier et al, here is the link: https://meet.google.com/fmt-dhdd-tzj

**Reply by darcymason** [reply]
@ftessier, @rtownson, As mentioned in the meeting, here is a link to the paper I saw for photon-only EGSnrc with GPU: https://jonaslippuner.com/publications/Lippuner_CUDA_EGS_2011_gray.pdf.  Still not sure if this was the same that you had mentioned.

**Comment by ftessier** [comment]
There is also an option to create `Polls` under `Discussions`, but it seems to be for literal question, not scheduling; I will send github this feature request! 🏄🏻

**Comment by SimonBiggs** [comment]
Hi @ftessier and @rtownson,

Great conversation today. 

Regarding the hardware choice. I see two options:

- 1st option -- Try and match the average consumer / contributor with a reasonably powered laptop
- 2nd option -- Try and match what a hospital might be interested in purchasing

If you want to go with the first option then the 3070 ti is a good choice. The inter-webs also tell me that the 4070 is coming out ~mid-May. But, even then, it'll probably still be more cost effective to go with something like the 3070 ti. Here it's probably worth hitting something with ~8 GB of VRAM.

If you want to go with the second option, and match a server that a hospital might be interested in purchasing then a machine with 2x3090 ti s would be appropriate. You could also go with the more recent 4090, but there are reports of instability, so me personally, I'd stick with the 3090 ti model for now. This equips the GPU compute with 48 GB of VRAM.

Here is the particular machine learning server specs that I built two years ago:
https://forums.fast.ai/t/recommendations-on-new-2-x-rtx-3090-setup/78202/274

Hardware like that would be quite appropriate, and if built like that, it has the side benefit of being a brilliant machine for AI as well.

Before purchasing, I'd recommend reaching out on that forum once you have your build decided. @balnazzar is exceptionally helpful and knowledgable.

Let me know if you'd like ssh access to the above machine and I can create a user account for you and give you access. That way both you and @darcymason will be using the same specced machine. You'll need to install everything for your own user though, as I am not allowing user accounts to have sudo access, nor are can they see/access the files of other users.

If you send me via email (me@simonbiggs.net) your public SSH key (say generated following [this guide](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)) I can give you remote access.

Regarding tooling once you are SSHed, here are some tips:

> Also, if you want to access a Jupyter Lab server on your local machine
> while hosting it on the server you can use the following:
> https://thedatafrog.com/en/articles/remote-jupyter-notebooks/
> 
> Also VSCode has a way to directly connect and edit files on a server via SSH:
> https://code.visualstudio.com/docs/remote/ssh
> 
> I haven't given you sudo access, but feel free to ask if you want
> anything installed and I can do that for you.
> 
> I tend to install Python within my home user using asdf:
> https://github.com/asdf-community/asdf-python
> 
> But you can feel free to install it however you'd like.
> 
> All the CUDA drivers are set up, so numba/pytorch/tensorflow GPU
> acceleration should all just work.
> 
> The following two tools I have installed will be really helpful for
> you I suspect:
> 
> - nvtop (live GPU monitoring)
> - htop (live CPU monitoring)

Cheers :),
Simon

**Reply by ftessier** [reply]
Thank you for all the information; please don't delete this discussion thread! I will send public key when I take you up on the offer. 🙏🏻
